python/test/unit/language/test_core.py

```diff
@@ -2169,12 +2169,12 @@ def kernel(X, stride_xm, stride_xk,
                Y, stride_yk, stride_yn,
                W, stride_wn, stride_wl,
                Z, stride_zm, stride_zn,
-               out_dtype: tl.constexpr,
                BLOCK_M: tl.constexpr, BLOCK_N: tl.constexpr, BLOCK_K: tl.constexpr,
                ADD_MATRIX: tl.constexpr, ADD_ROWS: tl.constexpr, ADD_COLS: tl.constexpr,
                ALLOW_TF32: tl.constexpr,
                DO_SOFTMAX: tl.constexpr, CHAIN_DOT: tl.constexpr,
-               COL_A: tl.constexpr, COL_B: tl.constexpr):
+               COL_A: tl.constexpr, COL_B: tl.constexpr,
+               out_dtype : tl.constexpr = tl.float32):
```

Comment:
Change to `out_dtype : tl.constexpr`? Since you specify `out_dtype` explicitly in the argument list.

Reply:
This needs to be explicit in the signature to trigger this kind of failure. The issue happens when fetching the default values from the signature using inspect library. Not passing a default value will not trigger the NameError failure.

Reply:
Does this make sense to resolve? @Jokeren